### PR TITLE
filter nonexistent cases from groups, and show warning icon

### PR DESCRIPTION
**Pull request by adamabeshouse**
for the screenshots, I just considered samples nonexistent at random

image

image

image

### CODE REVIEW

File: src/pages/groupComparison/GroupComparisonUtils.tsx [hunk 1]
```diff
@@ -279,4 +326,17 @@ export function getPieChartGroupFilters(
         });
     }
     return newFilters;
+}
+
+export function getMissingSamplesMessage(
```
Comment (alisman) [inline src/pages/groupComparison/GroupComparisonUtils.tsx]: we should learn signature of stateless components and use it

File: src/pages/groupComparison/comparisonGroupManager/ComparisonGroupManager.tsx [hunk 2]
```diff
@@ -53,7 +59,7 @@ export default class ComparisonGroupManager extends React.Component<IComparisonG
         await:()=>[this.props.store.comparisonGroups],
         invoke:()=>Promise.resolve(
             // TODO: fuzzy string search?
-            this.props.store.comparisonGroups.result!.filter(group=>group.data.name.toLowerCase().indexOf(this.groupNameFilter.toLowerCase()) > -1)
+            this.props.store.comparisonGroups.result!.filter(group=>group.name.toLowerCase().indexOf(this.groupNameFilter.toLowerCase()) > -1)
```
Comment (alisman) [inline src/pages/groupComparison/comparisonGroupManager/ComparisonGroupManager.tsx]: lets use RegExp(blah,"i").test  and get rid of all the lowercase and indexOf stuff

File: src/pages/groupComparison/styles.module.scss [hunk 3]
```diff
@@ -14,14 +14,9 @@
...
-        :global(div.groupItem):first-child {
-            margin-top:0px;
-        }
-        :global(div.groupItem):last-child {
+        :global(div.checkbox) {
             margin-bottom:0px;
+            margin-top:0px;
         }
 
         :global(.markedForDeletion) {
```
Comment (alisman) [inline src/pages/groupComparison/styles.module.scss]: i know this isn't part of this PR but seems like maybe "markedForDeletion" would be a good to not have global here.    I guess maybe groupItem is another.

Comment (adamabeshouse) [reply src/pages/groupComparison/styles.module.scss]: kk. I actually got rid of groupItem altogether

**Pull request merged**